src/shared/sam/cli/samCliBuildCommand.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { extensionSettingsPrefix } from '../../constants'
+import { fileExists } from '../../filesystemUtilities'
+import { DefaultSettingsConfiguration } from '../../settingsConfiguration'
+import { ChildProcess, ChildProcessResult } from '../../utilities/childProcess'
+import { SamCliCommand } from './samCliCommand'
+import { SamCliConfiguration } from './samCliConfiguration'
+import { DefaultSamCliLocationProvider } from './samCliLocator'
+
+export interface SamCliBuildCommandResponse {
+}
+
+export class SamCliBuildCommand extends SamCliCommand<SamCliBuildCommandResponse> {
+    private readonly _buildDir: string
+    private readonly _baseDir: string
+    private readonly _templatePath: string
+
+    public constructor(
+        buildDir: string,
+        baseDir: string,
+        templatePath: string,
+        config: SamCliConfiguration = new SamCliConfiguration(
+            new DefaultSettingsConfiguration(extensionSettingsPrefix),
+            new DefaultSamCliLocationProvider()
+        )) {
+        super(config)
+        this._buildDir = buildDir
+        this._baseDir = baseDir
+        this._templatePath = templatePath
+    }
+
+    public async execute(): Promise<SamCliBuildCommandResponse> {
+        await this.validate()
+
+        const childProcess: ChildProcess = new ChildProcess(
+            this.samCliLocation!,
```

Comment:
Minor: I don't like the implicit requirement of calling `this.validate`, then truthiness-asserting the location. What if `validate` returned the location as `string` (without `| undefined`)?

Reply:
I don't like the idea of overloading validate to return a specific value, but I can hide samCliLocation in the abstract class behind a getter that would throw if undefined